src/app/wallets/send-lightning.ts

```diff
@@ -370,22 +371,95 @@ const executePaymentViaIntraledger = async ({
   )
   if (validatedResult instanceof AlreadyPaidError) return PaymentSendStatus.AlreadyPaid
   if (validatedResult instanceof Error) return validatedResult
-  const { pubkey: recipientPubkey, walletId: recipientWalletId } = walletInvoice
+  const {
+    pubkey: recipientPubkey,
+    walletId: recipientWalletId,
+    cents: centsInvoice,
+  } = walletInvoice
 
   const recipientWallet = await WalletsRepository().findById(recipientWalletId)
   if (recipientWallet instanceof Error) return recipientWallet
 
-  const amountDisplayCurrency = dCConverter.fromSats(amountSats)
+  const dealerPriceService = DealerPriceService()
+
+  if (
+    senderWallet.currency === WalletCurrency.Usd &&
+    recipientWallet.currency === WalletCurrency.Btc &&
+    !invoiceWithAmount
+  ) {
+    // side effect: spread is paid by the recipient
+    cents = toCents(amount)
+    const sats_ = await dealerPriceService.getSatsFromCentsForImmediateSell(cents)
+    if (sats_ instanceof DealerPriceServiceError) return sats_
+    sats = sats_
+  } else if (
+    senderWallet.currency === WalletCurrency.Usd &&
+    recipientWallet.currency === WalletCurrency.Btc &&
+    invoiceWithAmount
+  ) {
+    // TODO: need to get an API for a quote
+    // so that a USD wallet can get a price, in USD, from a "probe", on a lightning invoice
+    sats = toSats(amount)
+    const cents_ = await dealerPriceService.getCentsFromSatsForImmediateSell(sats)
+    if (cents_ instanceof DealerPriceServiceError) return cents_
+    cents = cents_
+  } else if (
+    senderWallet.currency === WalletCurrency.Usd &&
+    recipientWallet.currency === WalletCurrency.Usd &&
+    !invoiceWithAmount
+  ) {
+    cents = toCents(amount)
+  } else if (
+    senderWallet.currency === WalletCurrency.Usd &&
+    recipientWallet.currency === WalletCurrency.Usd &&
+    invoiceWithAmount
+  ) {
+    cents = centsInvoice
+  } else if (
+    senderWallet.currency === WalletCurrency.Btc &&
+    recipientWallet.currency === WalletCurrency.Usd &&
+    !invoiceWithAmount
+  ) {
+    sats = toSats(amount)
+    const cents_ = await dealerPriceService.getCentsFromSatsForImmediateSell(sats)
+    if (cents_ instanceof DealerPriceServiceError) return cents_
+    cents = cents_
+  } else if (
+    senderWallet.currency === WalletCurrency.Btc &&
+    recipientWallet.currency === WalletCurrency.Usd &&
+    invoiceWithAmount
+  ) {
+    // side effect: spread is paid by the sender
+    cents = centsInvoice
+    sats = toSats(amount)
+  } else {
+    sats = toSats(amount)
+  }
+
+  const amountDisplayCurrency = sats
+    ? dCConverter.fromSats(sats)
+    : dCConverter.fromCents(cents!)
```

Comment:
Potential idea to get around typing issue here
```suggestion
const amountDisplayCurrency = sats
  ? dCConverter.fromSats(sats)
  : cents
  ? dCConverter.fromCents(cents)
  : new NotImplementedError("Should never get here")
if (amountDisplayCurrency instanceof Error) return amountDisplayCurrency
```

Reply:
could be an option, but still hacky IMO. I think we will want to refactor those use cases in different function so that we have a stronger typing, and those would not be necessary any more